pages/04_multitenancy_and_rls.md

```diff
@@ -0,0 +1,68 @@
+* If you have a separate database for each tenant, build a template dashboard and use the Document API to clone the dashboard for each tenant.
+* If your tenants share a database and there is filtering/entitlement logic to separate who should see which row, use user filters.
+* Use Initial SQL + in-database security if you already have in-database security set up *and* plan on using live connections to the database
+
+###  Template Dashboard + Document API
+[The Document API](https://github.com/tableau/document-api-python) is a Python API for modifying and querying Tableau content files (TWB/TWBX, TDS/TDSX). If your data environment is such that each tenant has their own database (and they are each structurally similar), then you can use the Document API to create a single workbook for each of your tenants. The workflow is:
+
+First, build a 'template' workbook that points at a single database. Build the necessary visualizations and dashboards.
+Then, create a script that:
+1) For each tenant, uses the Document API to change the database connection information, and save a copy. [Sample Script](https://github.com/tableau/document-api-python/blob/master/samples/replicate-workbook/replicate_workbook.py)
+2) Uses the [REST API (or Server Client Library)](./03_server_management_and_restapi.md)  to publish each tenant's workbook to Server.
+
+The above is all that is necessary if your workbooks have 'embedded datasources' (the datasources haven't been published separately). If you do choose to [publish the datasources to take advantage of Tableau Server's Data Server](https://onlinehelp.tableau.com/current/pro/desktop/en-us/publish_datasources.html), the Document API will still work. You will just have to apply the above workflow first to the datasource(s) and then to the workbooks:
+
+First, build a 'template' datasource, publish it, and build a 'template' workbook that points at the published datasource.
+Then, create a script that:
+1) For each tenant, uses the Document API to change the database connection information in the TDS (datasource file), and save a copy.
+2) Uses the [REST API (or Server Client Library)](./03_server_management_and_restapi.md) to publish each tenant's datasource to Server. The datasource will now have a connection string that you can use in the workbooks...
+3) For each tenant, for each workbook, uses the Document API to change the published datasource connection information, and save a copy.
+4) Uses the [REST API (or Server Client Library)](./03_server_management_and_restapi.md) to publish each all of the workbooks to Server.
+
+### User Filters
```

Comment:
Not really, unless you're using tickets on top of a 'service' account so every user impersonates a single user. I've added a comment in that page suggesting *not* doing that.